### PR TITLE
[fix](nereids)fix bug of rule twoPhaseAggregateWithCountDistinctMulti

### PR DESCRIPTION
**Pull request by starocean999**
# Proposed changes

Issue Number: close #xxx

## Problem summary
in AggregationNode, _merge_with_serialized_key_helper method should convert the key column to full column if the key column is null literal.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 34.54 seconds
 load time: 495 seconds
 storage size: 17122378838 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230120104731_clickbench_pr_84716.html

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/AggregateStrategies.java [hunk 1]
```diff
@@ -416,7 +416,9 @@ private List<PhysicalHashAggregate<Plan>> twoPhaseAggregateWithCountDistinctMult
                 .build();
         PhysicalHashAggregate<Plan> gatherLocalAgg = new PhysicalHashAggregate<>(
                 localAggGroupBy, localOutput, Optional.of(partitionExpressions),
-                new AggregateParam(AggPhase.LOCAL, AggMode.INPUT_TO_BUFFER),
+                new AggregateParam(AggPhase.LOCAL,
+                        logicalAgg.getGroupByExpressions().isEmpty() ? AggMode.INPUT_TO_RESULT
+                                : AggMode.INPUT_TO_BUFFER),
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/AggregateStrategies.java:421]: core with `select count(distinct k1, NULL), avg(k2) from test;`
maybe we set AggMode to INPUT_TO_RESULT, if and only if have no other agg function and have no group by exprs.

Comment (morrySnow) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/AggregateStrategies.java:421]: or, it must be three stage?

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by morrySnow — APPROVED** [review]

**Pull request merged**